pkg/scheduler/core/scheduling_queue.go

```diff
@@ -179,6 +193,10 @@ type PriorityQueue struct {
 	// pod was in flight (we were trying to schedule it). In such a case, we put
 	// the pod back into the activeQ if it is determined unschedulable.
 	receivedMoveRequest bool
+
+	// closed indicates that the queue is closed.
+	// It is mainly used to let Pop() exit its control loop while waiting for an item.
+	closed bool
```

Comment:
Not quite sure it's necessary to introduce another mutex to protect `closed` specifically.

Reply:
no, one mutex is enough to me :)

Reply:
Sweet.